File: palworld_info_server/src/items/entities/item.entity.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { PalItemEntity } from '../../pals/entities/pal_item.entity';
import { ItemRecipeEntity } from './item_recipe.entity';


export enum ItemType {
  'Weapon' = 'Weapon',
  'Special Weapon' = 'Special Weapon',
  'Armor' = 'Armor',
  'Accessory' = 'Accessory',
  'Material' = 'Material',
  'Consume' = 'Consume',
  'Ammo' = 'Ammo',
  'Food' = 'Food',
  'Essential' = 'Essential',
  'Glider' = 'Glider',
  'Monster Equip Weapon' = 'Monster Equip Weapon',
  'Blueprint' = 'Blueprint',
}

@Entity()
export class ItemEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ unique: true })
  name: string;

  @OneToMany(() => PalItemEntity, palItem => palItem.item)
  palItems: PalItemEntity[];

  @OneToMany(() => ItemRecipeEntity, itemRecipe => itemRecipe.material )
  recipe: ItemRecipeEntity;

  @OneToMany(() => ItemRecipeEntity, itemRecipe => itemRecipe.recipe )
  material: ItemRecipeEntity;

  @Column({ nullable: true })
  slug: string;

  @Column({ nullable: true })
  iconUrl: string;

  @Column({ nullable: true })
  summary: string;

  @Column({
    type: 'enum',
    enum: ItemType,
  })
  itemType: ItemType;


  @Column({ nullable: true })
  rank: number;

  @Column({ nullable: true })
  rarity: number;


  @Column({ nullable: true })
  price: number;

  @Column({ nullable: true, type: 'float' })
  weight: number;

  @Column({ nullable: true })
  maxStackCount: number;

  @Column({ nullable: true })
  physAttack: number;

  @Column({ nullable: true })
  durability: number;

  @Column({ nullable: true })
  restoreConcentration: number;

  @Column({ nullable: true })
  restoreSatiety: number;

  @Column({ nullable: true })
  passiveSkill: string;

}

File: palworld_info_app/lib/domains/item_entity.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:palworld_info_app/domains/pal_item_entity.dart';

part 'item_entity.freezed.dart';
part 'item_entity.g.dart';

@freezed
class ItemEntity with _$ItemEntity {

  const factory ItemEntity({
    int? id,
    String? name,
    String? slug,
    String? iconUrl,
    String? summary,
    String? itemType,
    int? rank,
    int? rarity,
    int? price,
    double? weight,
    int? maxStackCount,
    int? physAttack,
    int? durability,
    int? restoreConcentration,
    int? restoreSatiety,
    String? passiveSkill,
    List<PalItemEntity>? palItems,

  }) = _ItemEntity;

  factory ItemEntity.fromJson(Map<String, dynamic> json) =>
      _$ItemEntityFromJson(json);
}